pkg/scanner/local/scan.go

```diff
@@ -100,6 +101,76 @@ func (s Scanner) Scan(ctx context.Context, targetName, artifactKey string, blobK
 	return s.ScanTarget(ctx, target, options)
 }
 
+func postProcessApplications(apps []ftypes.Application, options types.ScanOptions) []ftypes.Application {
+	// Map to store nodejs lock file packages
+	nodeLockFilePackages := map[string]ftypes.Package{}
+	reqPHPPackages := make(map[string]struct{})
+	reqDevPHPPackages := make(map[string]struct{})
+
+	for i, app := range apps {
+		if len(app.Libraries) == 0 {
+			continue
+		}
+
+		isPkgSplitRequired := utils.IsPkgSplitRequired(app.Type)
+
+		// Get parents map for current target
+		parents := ftypes.Packages(app.Libraries).ParentDeps()
+
+		// Get app directory info for node
+		nodeAppDirInfo := utils.NodeAppDirInfo(app.FilePath)
+
+		for i, pkg := range app.Libraries {
+
+			// calculate rootDep
+			if len(parents) != 0 && (pkg.Indirect || isPkgSplitRequired) {
+				pkg.RootDependencies = utils.FindAncestor(pkg.ID, parents, map[string]struct{}{})
+				app.Libraries[i] = pkg
+
+				// if a pkg which direct and has atleast one root dependency
+				// we will consisder it as both direct and indirect
+				// so we append new entry for indirect and move root deps to the new entry
+				// note: for node-pkg this will done when we dedupe (at this point of time node-pkg won't have transitive info)
+				// splitting of vulnerabilties is handle automatically as vuln.detect operates on app.Libraries
```

Comment:
Would it make sense to elaborate the comment i.e since we splitting pkg A into pkg A (dir) and pkg A (indir), the corresponding vulnerabilities need to be highlighted twice in the scan report. Since vulnerabiltiies are fetched after post processing of applications, we'll automatically fetch vulnerabiltiies twice hence no splitting required.

Intention is to make it as clear as possible.